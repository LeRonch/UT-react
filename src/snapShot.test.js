
const FlightService = {
    list(){
      return {
        "arrivalDatetime": "2018-06-15T21:00:00",
        "departure": "DUB",
        "departureDatetime": "2018-06-15T17:25:00",
        "destination": "WRO",
        "flightNumber": "FR153",
      }
    }
  };
  
  describe('FlightService', () => {
      it('Should return a list of flights', () => {
          expect(FlightService.list()).toEqual({
            "arrivalDatetime": "2018-06-15T21:00:00",
            "departure": "DUB",
            "departureDatetime": "2018-06-15T17:25:00",
            "destination": "WRO",
            "flightNumber": "FR153",
          })
      });
  });
  
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };
  
  describe('User', () => {
      it('will fail every time', () => {
        expect(user).toMatchSnapshot();
      });

  });
  
  const user_2 = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };
  
  describe('User2', () => {
      it('will check the matchers and pass', () => {
        expect(user_2).toMatchSnapshot({
            createdAt: expect.any(Date),
            id: expect.any(Number),
        });
      });
  });
  
  const user_3 = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: Math.random().toString(36).substring(7),
  };
  
  describe('User3', () => {
      it('will check the matchers and pass', () => {
        expect(user_3).toMatchSnapshot(
            {
                createdAt: expect.any(Date),
                id: expect.any(Number),
                name: expect.any(String),
            }
          )
      });
  });