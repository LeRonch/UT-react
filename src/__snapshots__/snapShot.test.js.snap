// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User will fail every time 1`] = `
Object {
  "createdAt": 2022-06-09T12:55:57.145Z,
  "id": 4,
  "name": "LeBron James",
}
`;

exports[`User2 will check the matchers and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;

exports[`User3 will check the matchers and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": Any<String>,
}
`;
